contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -534,6 +550,12 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
             IVault(vaultAddress).mintForStrategy(mintForSwap);
         }
 
+        // approve the specific amount of WETH required
+        if (_swapWeth) {
+            IERC20(WETH).safeApprove(address(swapRouter), 0);
```

Comment:
WETH is a sane, compliant ERC20 token. Here's the approval code:

```
    function approve(address guy, uint wad) public returns (bool) {
        allowance[msg.sender][guy] = wad;
        emit Approval(msg.sender, guy, wad);
        return true;
    }
```

- We don't need to do the pre approval to zero, we can just set the approval we want
- We don't need to use safeApprove, WETH will always return a boolean true.

Reply:
thanks fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2266/commits/1321f22abb8223a30288c5d2a6549d4352ec0a69